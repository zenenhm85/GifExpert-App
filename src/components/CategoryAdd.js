import React, { useState } from "react";
import PropTypes from "prop-types";

export const CategoryAdd = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((categories) => {
        const find = categories.find((item) => item === inputValue);        
        if (!find) {
          setInputValue("");
          return [inputValue, ...categories];
          // setCategories([...categories,`Hunter ${categories.length}`]); This way is also good
        } else {            
            return categories;
        }
      });
      
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        className="input-categories"
        value={inputValue}
        onChange={handleInputChange}
        placeholder="Insert one category"
      />
    </form>
  );
};
CategoryAdd.propTypes = {
    setCategories: PropTypes.func.isRequired,
};