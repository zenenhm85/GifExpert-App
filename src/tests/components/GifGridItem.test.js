import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

import { GifGridItem } from "../../components/GifGridItem";

describe("Tests for GifGridItem", () => {
  const url = "https://image.jpg";
  const title = "Title of GIF";
  const wrapper = shallow(<GifGridItem url={url} title={title} />);

  test("should display the component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should have a paragraph with the title text", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });
  test("should have a img with url equal to this url and alt equal to this title", () => {
    const img = wrapper.find("img");
    /* expect(img.props().src).toBe(url);
        expect(img.props().alt).toBe(title); esto devuelve todas las propiedades */
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });
  test("the div should have the class animate__animated animate__fadeInDown animate__delay-2s", () => {
    const div = wrapper.find("div");

    const [
      ,
      animate__animated,
      animate__fadeInDown,
      animate__delay,
    ] = div.props().className.split(" ");
    expect(animate__animated).toBe("animate__animated");
    expect(animate__fadeInDown).toBe("animate__fadeInDown");
    expect(animate__delay).toBe("animate__delay-2s");
  });
});
