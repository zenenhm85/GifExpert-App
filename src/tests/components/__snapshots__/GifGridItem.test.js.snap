// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <GifGridItem />  should display the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeInDown animate__delay-2s"
>
  <img
    alt="Title of GIF"
    src="https://image.jpg"
  />
  <p>
    Title of GIF
  </p>
</div>
`;
